### PR TITLE
Top cards cruncher tests with slightly more obvious results.

### PR DESCRIPTION
**Pull request by grixa**
CC @dkorolev @mzhurovich

### CODE REVIEW

File: crunchers/test.cc [hunk 1]
```diff
@@ -201,16 +199,449 @@ TEST(CTFOCrunchersTest, ActiveUsersCruncherRemoteTest) {
...
+  EXPECT_EQ(calculators[0](top_cards.value[0][0]), top_cards.value[0][0].rate);
+  EXPECT_EQ(2.0, top_cards.value[0][0].rate);
+  // Cruncher #2 (2ms time window).
+  // Card #1: rate = SKIP (1x) / SEEN (1x) = 1.
+  ASSERT_EQ(1u, top_cards.value[1].size());
+  EXPECT_EQ(1u, top_cards.value[1][0].cid);
+  EXPECT_EQ(calculators[1](top_cards.value[1][0]), top_cards.value[1][0].rate);
+  EXPECT_EQ(1.0, top_cards.value[1][0].rate);
+
+  // Add one ctfo event for Card #2.
```
Comment (mzhurovich) [inline crunchers/test.cc]: s/ctfo/CTFO/

File: crunchers/test.cc [hunk 2]
```diff
@@ -201,16 +199,449 @@ TEST(CTFOCrunchersTest, ActiveUsersCruncherRemoteTest) {
...
+  EXPECT_EQ(3u, top_cards.value[0][0].cid);
+  EXPECT_EQ(calculators[0](top_cards.value[0][0]), top_cards.value[0][0].rate);
+  EXPECT_EQ(3.0, top_cards.value[0][0].rate);
+  EXPECT_EQ(2u, top_cards.value[0][1].cid);
+  EXPECT_EQ(calculators[0](top_cards.value[0][1]), top_cards.value[0][1].rate);
+  EXPECT_EQ(1.0, top_cards.value[0][1].rate);
+  // Cruncher #2 (2ms time window).
+  // Card #3: rate = (TFU (1x) + FAV (1x)) / SEEN (1x) = 2.
+  // Card #1: rate = SKIP (1x) / SEEN (1x) = 1.
+  // Card #2: rate = CTFO (1x) / SEEN (0x) = 0.
```
Comment (mzhurovich) [inline crunchers/test.cc:367]: You don't actually divide by zero, right? ;)

File: crunchers/test.cc [hunk 3]
```diff
@@ -201,16 +199,449 @@ TEST(CTFOCrunchersTest, ActiveUsersCruncherRemoteTest) {
...
+  ASSERT_EQ(1u, top_cards.value[0].size());
+  EXPECT_EQ(1u, top_cards.value[0][0].cid);
+  EXPECT_EQ(calculators[0](top_cards.value[0][0]), top_cards.value[0][0].rate);
+  EXPECT_EQ(2.0, top_cards.value[0][0].rate);
+  // Cruncher #2 (2ms time window).
+  // Card #1: rate = SKIP (1x) / SEEN (1x) = 1.
+  ASSERT_EQ(1u, top_cards.value[1].size());
+  EXPECT_EQ(1u, top_cards.value[1][0].cid);
+  EXPECT_EQ(calculators[1](top_cards.value[1][0]), top_cards.value[1][0].rate);
+  EXPECT_EQ(1.0, top_cards.value[1][0].rate);
```
Comment (mzhurovich) [inline crunchers/test.cc]: BTW `EXPECT_DOUBLE_EQ` is generally more right way to compare doubles.

File: crunchers/test.cc [hunk 4]
```diff
@@ -201,16 +199,449 @@ TEST(CTFOCrunchersTest, ActiveUsersCruncherRemoteTest) {
...
+  EXPECT_EQ(calculators[0](top_cards.value[0][0]), top_cards.value[0][0].rate);
+  EXPECT_EQ(2.0, top_cards.value[0][0].rate);
+  // Cruncher #2 (2ms time window).
+  // Card #3: rate = (TFU (1x) + FAV (1x) + CTFO (1x)) / SEEN (2x) = 1.5.
+  // Cards #1 and #2 are out of the time window.
+  ASSERT_EQ(1u, top_cards.value[1].size());
+  EXPECT_EQ(3u, top_cards.value[0][0].cid);
+  EXPECT_EQ(calculators[1](top_cards.value[1][0]), top_cards.value[1][0].rate);
+  EXPECT_EQ(1.5, top_cards.value[1][0].rate);
+}
```
Comment (mzhurovich) [inline crunchers/test.cc:633]: Can we get rid of copy-paste here and put all the tests in a separate function/class that is used for both local and remote subscriptions?

**Review by mzhurovich — COMMENTED** [review]
@grixa Thanks! These new tests are realy good!

File: crunchers/test.cc (hunk 4 above)
Comment (grixa) [reply crunchers/test.cc:633]: Of course. I suggest to do it for all of the crunchers tests at once in a separate pull request.

File: crunchers/test.cc (hunk 2 above)
Comment (grixa) [reply crunchers/test.cc:367]: Right. See a0fad9fc

File: crunchers/test.cc (hunk 4 above)
Comment (grixa) [reply crunchers/test.cc:633]: Done in #79, but this pull request should be merged before.

**Comment by grixa** [comment]
@mzhurovich how about to merge this request before the #79 ?

**Pull request merged**